### PR TITLE
dockerize OpenQA webui and worker

### PR DESCRIPTION
**Pull request by garretraziel**
Me and Josef Skládanka (jskladan@redhat.com) have been working on dockerizing OpenQA last few weeks. It makes development and deployment really easy for us at Fedora QA. It enables us to upgrade painlessly and deploy it into our Fedora infrastructure and it also really simplifies installation process. We think that a lot of other people would benefit from this, so we would love to make these Dockerfiles official.

Included is detailed README how to build and deploy these images.

### CODE REVIEW

File: docker/openqa_data/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM opensuse:13.2
+MAINTAINER Jan Sedlak <jsedlak@redhat.com> Josef Skladanka <jskladan@redhat.com>
+
+RUN zypper --non-interactive --gpg-auto-import-keys in ca-certificates git wget
```
Comment (lnussel) [inline docker/openqa_data/Dockerfile]: --gpg-auto-import-keys disables a major feature of the package manager. I don't  think you want to do that for production.

File: docker/webui/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    a2enmod proxy_wstunnel && \
+    a2enmod ssl && \
+    a2enflag SSL
+ADD openqa-ssl.conf /etc/apache2/vhosts.d/openqa-ssl.conf
+ADD openqa.conf /etc/apache2/vhosts.d/openqa.conf
+
+# set-up shared data and configuration
+RUN rm -rf /etc/openqa/openqa.ini /etc/openqa/client.conf \
+      /var/lib/openqa/share/factory /var/lib/openqa/share/tests \
+      /var/lib/openqa/db/db.sqlite /var/lib/openqa/testresults && \
```
Comment (lnussel) [inline docker/webui/Dockerfile]: if you create those directories as symlinks before actually installing the packages rpm would just follow the links so you'd not have to recreate everything yourself

File: docker/openqa_data/Dockerfile (hunk 1 above)
Comment (garretraziel) [inline docker/openqa_data/Dockerfile]: It's not necessary to use --gpg-auto-import-keys here, but it's necessary to use it when installing OpenQA. I don't know how to handle certificates of OpenQA repo (where to get that certificate and how to import it).

Comment (lnussel) [inline docker/openqa_data/Dockerfile]: curl https://build.opensuse.org/source/devel:openQA/_pubkey

You can use rpm --import on that

**Comment by garretraziel** [comment]
I'll change dockerfiles accordingly. I have also discovered that there is conflict between some Perl packages in Perl devel repo and official updates repo. I'll try to investigate it.

**Comment by lnussel** [comment]
devel:languages:perl should not be necessary btw. If you need it it's a bug

**Comment by garretraziel** [comment]
When I try to install openQA without this repo, it shows:

    Problem: nothing provides perl(Mojolicious::Plugin::Bootstrap3) needed by openQA-common-4.1-3.8.noarch

**Comment by lnussel** [comment]
hmm, the repository config in obs suggests that you also need to add the devel:openQA repo even if you want to install the stable version. You can use "zypper in --from devel_openQA_stable openQA" to request the specific openQA package from the stable repo. zypp's vendor lock should prevent it from upgrading to the unstable openQA later I think.

**Comment by aaannz** [comment]
I added that to project config to fix stable builds (see #320), I didn't want to copy/link packages. But obviously not perfect solution.

**Comment by coolo** [comment]
the project config won't help here as you still need the package installed into your system/docker. I guess we need to split devel:openQA into unstable openQA/os-autoinst and dependencies possibly needed also for stable builds

File: docker/openqa_data/Dockerfile (hunk 1 above)
Comment (garretraziel) [inline docker/openqa_data/Dockerfile]: I have opened an issue for this in openSUSE/docker-containers repo here: https://github.com/openSUSE/docker-containers/issues/3.

Docker image doesn't include any certificates, so user can't curl from https and check if server can be trusted. Another problem is that image also doesn't have GPG keys of standard repositories imported, so user is also unable to install `ca-certificates` or `curl` packages without passing `--gpg-auto-import-keys`.

Currently, only solution is to run: `zypper --non-interactive --gpg-auto-import-keys in ca-certificates curl` at the beginning.

**Pull request merged**